browser/brave_ads/search_result_ad/search_result_ad_handler.cc

```diff
@@ -0,0 +1,293 @@
+      ad_property->values->get_string_values().size() != 1) {
+    return false;
+  }
+
+  *out_value = ad_property->values->get_string_values().front();
+
+  return true;
+}
+
+bool GetInt64Value(const schema_org::mojom::PropertyPtr& ad_property,
+                   int64_t* out_value) {
+  DCHECK(ad_property);
+  DCHECK(out_value);
+
+  // Wrong attribute type
+  if (!ad_property->values->is_long_values() ||
+      ad_property->values->get_long_values().size() != 1) {
+    return false;
+  }
+
+  *out_value = ad_property->values->get_long_values().front();
+
+  return true;
+}
+
+bool GetDoubleValue(const schema_org::mojom::PropertyPtr& ad_property,
+                    double* out_value) {
+  DCHECK(ad_property);
+  DCHECK(out_value);
+
+  // Wrong attribute type
+  if (!ad_property->values->is_string_values() ||
+      ad_property->values->get_string_values().size() != 1) {
+    return false;
+  }
+
+  std::string value = ad_property->values->get_string_values().front();
+  return base::StringToDouble(value, out_value);
+}
+
+bool SetSearchAdProperty(const schema_org::mojom::PropertyPtr& ad_property,
+                         SearchResultAd* search_result_ad) {
+  DCHECK(ad_property);
+  DCHECK(search_result_ad);
+  DCHECK(search_result_ad->conversion);
+
+  const std::string& name = ad_property->name;
+  if (name == kDataPlacementId) {
+    return GetStringValue(ad_property, &search_result_ad->placement_id);
+  } else if (name == kDataCreativeInstanceId) {
+    return GetStringValue(ad_property, &search_result_ad->creative_instance_id);
+  } else if (name == kDataCreativeSetId) {
+    return GetStringValue(ad_property, &search_result_ad->creative_set_id);
+  } else if (name == kDataCampaignId) {
+    return GetStringValue(ad_property, &search_result_ad->campaign_id);
+  } else if (name == kDataAdvertiserId) {
+    return GetStringValue(ad_property, &search_result_ad->advertiser_id);
+  } else if (name == kDataLandingPage) {
+    return GetStringValue(ad_property, &search_result_ad->target_url);
+  } else if (name == kDataHeadlineText) {
+    return GetStringValue(ad_property, &search_result_ad->headline_text);
+  } else if (name == kDataDescription) {
+    return GetStringValue(ad_property, &search_result_ad->description);
+  } else if (name == kDataRewardsValue) {
+    return GetDoubleValue(ad_property, &search_result_ad->value);
+  } else if (name == kDataConversionTypeValue) {
+    return GetStringValue(ad_property, &search_result_ad->conversion->type);
+  } else if (name == kDataConversionUrlPatternValue) {
+    return GetStringValue(ad_property,
+                          &search_result_ad->conversion->url_pattern);
+  } else if (name == kDataConversionAdvertiserPublicKeyValue) {
+    return GetStringValue(ad_property,
+                          &search_result_ad->conversion->advertiser_public_key);
+  } else if (name == kDataConversionObservationWindowValue) {
+    return GetInt64Value(ad_property,
```

Comment:
Changed from int64 to int32 in latest https://github.com/brave/brave-core/pull/12467